eng/versioning/pom_file_version_scanner.ps1

```diff
@@ -37,7 +37,7 @@ $ValidParents = ("azure-sdk-parent", "azure-client-sdk-parent", "azure-data-sdk-
 # The problem with this is, it's a BOM file and the spring dependencies are pulled in through that which means any
 # dependencies may or may not have versions. Unfortunately, there are still version tags azure sdk client libraries
 # which means these files have to be "sort of" scanned.
-$SpringSampleParents = ("spring-boot-starter-parent", "azure-spring-boot-test-parent")
```

Comment:
Why are we removing this?

Reply:
The `azure-spring-boot-test-parent` was removed because there was no version added for the `stater` in the pom.xml of IT, and an exception would be thrown when the `pom_file_version_scanner.ps1` script was executed.
![image](https://user-images.githubusercontent.com/71206407/114142158-25bf0180-9945-11eb-91bd-9a85d9a75d63.png)

Reply:
I see... So I am afraid we can't merge this PR now. Since the versions in azure-spring-boot-bom aren't managed by the version script now, so it's possible that versions applied to our testing projects are not the same as defined in version_client.txt.